Doc/library/typing.rst

```diff
@@ -2631,6 +2631,13 @@ Functions and decorators
       x = reveal_type(1)  # prints "Runtime type is int"
       print(x)  # prints "1"
 
+   Note that the runtime type inferred by ``reveal_type()`` can be different
+   from other static type checkers. Static type checkers in some cases will
```

Comment:
Two points here:

- `reveal_type` at runtime doesn't really "infer" a type in the same way that static type checkers do; it just returns the _actual_ type at runtime
- The phrase "other type checkers" is confusing, as it implies that the CPython runtime is a type checker, which it isn't. (It does do "type checking" in various functions for various reasons, but that doesn't make it a "type checker".)

Reply:
Updated accordingly.